### ISSUE TITLE
Rework of Source/Item property handlers

### ISSUE THREAD

**Issue by Matrim-Cauthon** [xjsframework/xjs#111]
Based on our internal discussions, we will revisit the API split between `Source` and `Item`. We will go with the general spirit of the suggestion in #109 but specific implementation notes follow:
1. We _will_ "undeprecate" `Source`. In order to clearly show that some properties belong only to sources (such as Browser Size), we will add those methods to `Source`. (But see additional concerns at the bottom.)
2. We _will_ retain source-specific properties in `Item`. Underneath the hood, our components will just bubble up the property changes to the source level so this will not break existing code.
3. For the above, we _will_ add warnings when source-specific properties are being called from `Item`. This is in line with our efforts towards plugin developer education regarding the source and item properties.
4. We _will not_ implement `Scene.getSources` to get `Source` instances, because we want to make it clear   on the API level that sources do not belong to a scene; the video item or items linked to them are the elements that belong in scenes. This means that our current deprecation of `Scene.getSources` will be retained. However, we _will_ improve the docs there.
5. We _may_ implement `App.getSources` in line with the above point. This method will return the complete set of sources in the entire application. (See below.)
6. We _must_ fix our documentation in the next release. This'll be tackled in a separate issue in our website's Github.

With the above points, I believe there are still a couple of issues we need to address before we start implementation.

For (1), we need to review our implementation of the old deprecated features of `Source`. Specifically, `Source.getCurrentSource`. For compatibility purposes, in 1.4.0 we simply returned the first item in the item list (this was under the assumption that pre-xjs@1.4.0, global sources didn't exist; thus, all plugins had exactly one linked item.) We need to make an important decision: do we return the actual `Source`, or retain current behavior? Note that this may possibly lead to breaking changes.

For (5), I think we just need to consider sources in the preview editor but not in the main view, and other such edge cases, so we can go forward with the implementation.

**Comment by virn** [comment]
For (1) in undeprecating Source: I have made the following changes.
1. Created an iSource which holds Source specific methods, which will also be retained in Item.
2. Added a checker **isItemCall** on Item and Source which will be checked on iSource and log a warning if Source methods  are called on Item.
3. Created separate i(Html/VideoPlaylist/etc)Source specific classes to hold Source specific methods (BrowserSize/BrowserRightClick/etc) which is extended through mixins to the main class(html/videoplaylist/media). These classes are also used for checking **isItemCall** and log the warning.

The initial implementation on this can be seen on Item, Source and iSource, and HtmlItem, HtmlSource and iHtmlSource.
